apps/assets/migrations/0016_auto_20250331_1149.py

```diff
@@ -0,0 +1,166 @@
+                  "setting": {
+                      "sftp_enabled": true,
+                      "sftp_home": "/tmp"
+                  }
+              },
+              {
+                  "name": "vnc",
+                  "port": 5900,
+                  "primary": false,
+                  "required": false,
+                  "default": false,
+                  "public": true,
+                  "setting": {}
+              },
+              {
+                  "name": "winrm",
+                  "port": 5985,
+                  "primary": false,
+                  "required": false,
+                  "default": false,
+                  "public": false,
+                  "setting": {
+                      "use_ssl": false
+                  }
+              }
+          ]
+      }
+    ]
+    """
+    platform_model = apps.get_model('assets', 'Platform')
+    automation_cls = apps.get_model('assets', 'PlatformAutomation')
+    platform_datas = json.loads(data)
+
+    for platform_data in platform_datas:
+        AllTypes.create_or_update_by_platform_data(platform_data, platform_cls=platform_model,
+                                                   automation_cls=automation_cls)
+
+
+class Migration(migrations.Migration):
+    dependencies = [
+        ("assets", "0015_automationexecution_type"),
+    ]
+
+    operations = [
+        migrations.RunPython(add_ad_host_type),
+        migrations.CreateModel(
+            name="AD",
+            fields=[
+                (
+                    "asset_ptr",
+                    models.OneToOneField(
+                        auto_created=True,
+                        on_delete=django.db.models.deletion.CASCADE,
+                        parent_link=True,
+                        primary_key=True,
+                        serialize=False,
+                        to="assets.asset",
+                    ),
+                ),
+                (
+                    "domain_name",
+                    models.CharField(
+                        blank=True,
+                        default="",
+                        max_length=128,
+                        verbose_name="Domain name",
+                    ),
+                ),
+            ],
+            options={
+                "verbose_name": "Active Directory",
+            },
+            bases=("assets.asset",),
+        ),
+        migrations.AddField(
+            model_name="platform",
+            name="ad",
+            field=models.ForeignKey(
+                blank=True,
+                null=True,
+                on_delete=django.db.models.deletion.SET_NULL,
+                related_name="ad_platforms",
+                to="assets.ad",
+                verbose_name="Active Directory",
+            ),
+        ),
+        migrations.AddField(
+            model_name="platform",
+            name="ad_enabled",
+            field=models.BooleanField(default=False, verbose_name="AD enabled"),
+        ),
+    ]
```

Comment:
The given code seems to be a migration file that was run on Apache Airflow using Django ORM. The main difference between the source and target versions is related to the automation execution type, which in the old version had `Type="all"` set.

In the new version:

```
    AutomationExecutionType = {
        Type("ad"):
```